src/wap_client.c

```diff
@@ -211,7 +221,8 @@ signal_have_save_ok (client_t *self)
 static void
 signal_have_start_ok (client_t *self)
 {
-    zsock_send (self->cmdpipe, "si", "START OK", 0);
+    zsock_send (self->cmdpipe, "sii", "START OK", 0,
+                wap_proto_curr_height (self->message));
 }
```

Comment:
Couldn't the system figure this out on its own? If I added another field to be returned along with `curr_height` in the daemon, I would have to pass it here to `zsock_send` too.

The wallet developer would have to act upon such changes in the message formats (extra field, etc.) in two places: one here and another in the real wallet code where the response is actually made use of by the wallet.

I feel the same way about `prepare_start_command` too. If the `start` message format as predecided, is to contain a "start_height" field, shouldn't the underlying system be able to take care of everything (like storing it in a message) after a `wap_client_start` call?

Of course, I'm not very knowledgeable about the internals, but that would've been simpler usage IMO, unless there is a benefit of such finer grain control.